scripts/checkstyle.py

```diff
@@ -140,6 +157,20 @@ def main():
             ],
             'found': 0
         },
+        {
+            'name': 'no-function-keyword',
+            'regex': '^[ \\t]*(?P<match>function .*?(?:\\([ \\t]*\\))?[ \\t]*){',
+            'reason': 'Only allow functions declared like `fn_name() { :; }` for consistency',
```

Comment:
Consistency, sure, but a better reasoning is because it is more portable. `function` is not as portable. Shellcheck has a rule for this https://www.shellcheck.net/wiki/SC2113 It is just not showing as an error because it only applies to `--shell sh` not `--shell bash` or `--shell bats`.

Reply:
Yeah, `function` isn't portable, but since checkstyle only scans Bash/Bats files, I guess I thought it would be better to put styling as a reason. I will change it to add a reference to the SC error too

Reply:
Yes please. Consistency & portability (as pointed out by Shellcheck in SC2113)